File: content/LLD/Reference-Code/9. State Pattern/transport_Service.py
from transport_mode import TransportMode


class TransportService:
    def __init__(self, mode: TransportMode):
        self.__mode: TransportMode = mode

    def set_mode(self, new_mode: TransportMode):
        self.__mode: TransportMode = new_mode

    def eta(self):
        self.__mode.eta()

    def directions(self):
        self.__mode.directions()

File: content/LLD/Reference-Code/9. State Pattern/TransportService.java
public class TransportService {
    private TransportMode mode;

    public TransportService(TransportMode mode) {
        this.mode = mode;
    }

    public void setMode(TransportMode newMode) {
        this.mode = newMode;
    }

    public void eta() {
        this.mode.eta();
    }

    public void directions() {
        this.mode.directions();
    }
}
